main.go

```diff
@@ -357,7 +361,52 @@ func notifyWaiters(w http.ResponseWriter, r *http.Request) {
 }
 
 func main() {
+	log.SetFlags(log.Lshortfile | log.Ldate | log.Lmicroseconds)
 	flag.Parse()
-	http.HandleFunc("/", handle)
-	log.Fatal(http.ListenAndServe(*flagHTTP, nil))
+
+	tmpDir, err := ensureTmpOutputDir()
+	if err != nil {
+		log.Fatalf("Failed to create temporary directory: %v", err)
+	}
+
+	defer os.RemoveAll(tmpDir)
+
+	var server http.Server
+
+	idleConnsClosed := make(chan struct{})
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, os.Interrupt)
+		<-sigint
+
+		log.Printf("Shutting down server...")
+
+		// We received an interrupt signal, shut down.
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
```

Comment:
Just out of curiosity, why 10[ms]?

Reply:
10ms is an arbitrary choice, but if just context.Background() is used by itself then the server never shuts down because the web browser maintains a connection to the server I think? 10ms just forces the server to close basically.

Reply:
Sure. I'm fine we know this value is just arbtrary.